specutils/io/default_loaders/wcs_fits.py

```diff
@@ -85,9 +90,80 @@ def wcs1d_fits_loader(file_obj, spectral_axis_unit=None, flux_unit=None,
     if not isinstance(file_obj, fits.hdu.hdulist.HDUList):
         hdulist.close()
 
+        if wcs.naxis > 4:
+            raise ValueError('FITS file input to wcs1d_fits_loader is > 4D')
+        elif wcs.naxis > 1:
+            for i in range(wcs.naxis - 1, 0, -1):
+                try:
+                    wcs = wcs.dropaxis(i)
+                except(_wcs.NonseparableSubimageCoordinateSystemError) as e:
+                    raise ValueError(f'WCS cannot be reduced to 1D: {e} {wcs}')
+
     return Spectrum1D(flux=data, wcs=wcs, meta=meta)
 
 
+@custom_writer("wcs1d-fits")
+def wcs1d_fits_writer(spectrum, file_name, update_header=False, **kwargs):
+    """
+    Write spectrum with spectral axis defined by its WCS to (primary)
+    IMAGE_HDU of a FITS file.
+
+    Parameters
+    ----------
+    spectrum: Spectrum1D
+    file_name: str
+        The path to the FITS file
+    update_header: bool
+        Update FITS header with all compatible entries in `spectrum.meta`
+    unit : str or `~astropy.units.Unit`
+        Unit for the flux (and associated uncertainty)
+    dtype : str or `~numpy.dtype`
+        Floating point type for storing flux array
+    """
+    # Create HDU list from WCS
+    try:
+        wcs = spectrum.wcs
+        hdulist = wcs.to_fits()
+        header = hdulist[0].header
+    except AttributeError as err:
+        raise ValueError(f'Only Spectrum1D objects with valid WCS can be written as wcs1d: {err}')
+
+    # Verify spectral axis constructed from WCS
+    wl = spectrum.spectral_axis
+    dwl = (wcs.all_pix2world(np.arange(len(wl)), 0) - wl.value) / wl.value
+    if np.abs(dwl).max() > 1.e-10:
+        m = np.abs(dwl).argmax()
+        raise ValueError('Relative difference between WCS spectral axis and'
+                         f'spectral_axis at {m:}: dwl[m]')
+
+    if update_header:
+        hdr_types = (str, int, float, complex, bool,
+                     np.floating, np.integer, np.complexfloating, np.bool_)
+        header.update([keyword for keyword in spectrum.meta.items() if
+                       (isinstance(keyword[1], hdr_types) and
+                        keyword[0] not in ('NAXIS', 'NAXIS1', 'NAXIS2'))])
+
+    # Cannot include uncertainty in IMAGE_HDU - maybe provide option to
+    # separately write this to BINARY_TBL extension later.
+    if spectrum.uncertainty is not None:
+        message = "Saving uncertainties in wcs1d format is not yet supported!"
+        warnings.warn(message, AstropyUserWarning)
+
+    # Add flux array and unit
+    ftype = kwargs.pop('dtype', spectrum.flux.dtype)
+    funit = u.Unit(kwargs.pop('unit', spectrum.flux.unit))
+    flux = spectrum.flux.to(funit, equivalencies=u.spectral_density(wl))
+    hdulist[0].data = flux.value.astype(ftype)
+
+    if hasattr(funit, 'long_names') and len(funit.long_names) > 0:
```

Comment:
Just a clarification: does `long_names` give something different from `physical_type`? In my (limited testing), `physical_type` == `long_names[1]`.

Reply:
`long_names` can include any (alias) names the unit was given on initialisation; it rarely (if ever) includes `physical_type`. E.g.
```
>>> u.Jy.long_names
['Jansky', 'jansky']
>>> u.Jy.physical_type
'spectral flux density'
```
so `comment` would become `'[Jansky] spectral flux density'` for `BUNIT = Jy` (`u.Jy.to_string()`).

Reply:
Ah, sorry, I got my lines (of code) crossed: I meant the does `long_names` give something different from `to_string`; but was testing with units that seem to be inconsistent (e.g. Angstroms gives `['AA', 'angstrom']` for long names, and `angstrom` for `to_string()`, but centimeters gives `['centimeter']` and `cm` respectively).

Reply:
Yes, `short_names` and `long_names` for `u.AA` don't seem to give terribly intuitive results...